src/main/java/com/lambdaworks/redis/resource/Delay.java

```diff
@@ -51,46 +58,159 @@ public TimeUnit getTimeUnit() {
-        return new ExponentialDelay(lower, upper, unit, powersOf);
+        final Delay statelessDelay = new ExponentialDelay(lower, upper, unit, powersOf);
+        return () -> statelessDelay; // Reuse this because it's stateless.
+    }
+
+    /**
+     * Creates a new {@link EqualJitterDelay} supplier with default boundaries.
+     *
+     * @return a created {@link Supplier<EqualJitterDelay>}.
+     */
+    public static Supplier<Delay> equalJitter() {
+        return equalJitter(DEFAULT_LOWER_BOUND, DEFAULT_UPPER_BOUND, 1L, DEFAULT_TIMEUNIT);
+    }
+
+    /**
+     * Creates a new {@link EqualJitterDelay} supplier.
+     *
+     * @param lower the lower boundary, must be non-negative
+     * @param upper the upper boundary, must be greater than the lower boundary
+     * @param base the base, must be greater or equal to 1
+     * @param unit the unit of the delay.
+     * @return a created {@link Supplier<EqualJitterDelay>}.
+     */
+    public static Supplier<Delay> equalJitter(long lower, long upper, long base, TimeUnit unit) {
+
+        LettuceAssert.isTrue(lower >= 0, "Lower boundary must be greater or equal to 0");
+        LettuceAssert.isTrue(upper > lower, "Upper boundary must be greater than the lower boundary");
+        LettuceAssert.isTrue(base >= 1, "Base must be greater or equal to 1");
+
+        final Delay statelessDelay = new EqualJitterDelay(lower, upper, base, unit);
+        return () -> statelessDelay; // Reuse this because it's stateless.
+    }
+
+    /**
+     * Creates a new {@link FullJitterDelay} supplier with default boundaries.
+     *
+     * @return a created {@link Supplier<FullJitterDelay>}.
+     */
+    public static Supplier<Delay> fullJitter() {
+        return fullJitter(DEFAULT_LOWER_BOUND, DEFAULT_UPPER_BOUND, 1L, DEFAULT_TIMEUNIT);
+    }
+
+    /**
+     * Creates a new {@link FullJitterDelay} supplier.
+     *
+     * @param lower the lower boundary, must be non-negative
+     * @param upper the upper boundary, must be greater than the lower boundary
+     * @param base the base, must be greater or equal to 1
+     * @param unit the unit of the delay.
+     * @return a created {@link Supplier<FullJitterDelay>}.
+     */
+    public static Supplier<Delay> fullJitter(long lower, long upper, long base, TimeUnit unit) {
+
+        LettuceAssert.isTrue(lower >= 0, "Lower boundary must be greater or equal to 0");
+        LettuceAssert.isTrue(upper > lower, "Upper boundary must be greater than the lower boundary");
+        LettuceAssert.isTrue(base >= 1, "Base must be greater or equal to 1");
+
+        final Delay statelessDelay = new FullJitterDelay(lower, upper, base, unit);
+        return () -> statelessDelay; // Reuse this because it's stateless.
+    }
+
+    /**
+     * Creates a new {@link DecorrelatedJitterDelay} supplier with default boundaries.
+     *
+     * @return a created {@link Supplier<DecorrelatedJitterDelay>}.
+     */
+    public static Supplier<Delay> decorrelatedJitter() {
```

Comment:
This works nicely with `ClientResourcesBuilder` so always the right method is picked when calling `builder.reconnectDelay(Delay. decorrelatedJitter())`. Stateless delays are OK to share while the stateful should be a `Supplier`.